software/test_project/src/map_builder/ElementWall.java

```diff
@@ -21,33 +14,23 @@ public ElementWall(int gridX, int gridY, Color elementColor){
 
 	@Override
 	public void draw(Graphics graphics) {
-//		System.out.println("okay drawing specific ElementWall");
-		//draw fill
-//		graphics.setColor(getColor());
-//		graphics.fillRect(getMapX(), getMapY(), getWidth(), getHeight());
-
-		//draw border
-		//			graphics.setColor(Color.DARK_GRAY);
-//		graphics.drawRect(getMapX(), getMapY(), getWidth(), getHeight());
-		
-		graphics.setColor(getColor());
-		graphics.fillRect(
-				(getGridX() * Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN, 
-				(getGridY()*Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN+Constants.WINDOW_HEADER_HEIGHT, 
-				getWidth(), getHeight());
-		
-		// draw border
-		// set custom color here with graphics.setColor(borderColor);
-		graphics.drawRect(
-				(getGridX() * Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN, 
-				(getGridY()*Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN+Constants.WINDOW_HEADER_HEIGHT, 
-				getWidth(), getHeight());
+		int x = (getGridX() * Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN;
+		int y = (getGridY()*Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN+Constants.WINDOW_HEADER_HEIGHT;
+
+		Graphics2D g2d = (Graphics2D)graphics;
+
+		g2d.setColor(getColor());
+		g2d.fillRect(x, y, getWidth(), getHeight());
+
+
+		//Border color is just the main color but with an alpha value
+		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),85);
```

Comment:
Maybe consider declaring a variable for the alpha value either on local class level or on global level if this alpha value should also be used for other elements...
```suggestion
                int alpha = 85;
                //...
		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),alpha);
```